xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -3447,11 +3459,153 @@ public void readObjectsFromForm(EditForm eform, XWikiContext context) throws XWi
         }
     }
 
+    /**
+     * Generate a map from the request parameters of the form '<spacename>.<classname>_<number>_<propertyname>' Keys of
+     * this map will be the reference '<spacename>.<classname>' to the Class (for example, 'XWiki.XWikiRights'), the
+     * content is a list where each element describe property for the object <number>. Element of the list is a map
+     * where key is <propertyname> and content is the array of corresponding values.
+     * 
+     * Example with a list of HTTP parameters:
+     * 
+     * * XWiki.XWikiRights_0_users=XWiki.Admin
+     * * XWiki.XWikiRights_0_users=XWiki.Me
+     * * XWiki.XWikiRights_0_groups=XWiki.XWikiAllGroup
+     * * XWiki.XWikiRights_1_user=XWiki.Admin
+     * * XWiki.XWikiUsers_1_name=Spirou
+     * 
+     * will result in the following map
+     * 
+     * {
+     *   "XWiki.XWikiRights": {
+     *     "0": {
+     *       "users": ["XWiki.Admin", "XWiki.Me"],
+     *       "groups": ["XWiki.XWikiAllGroup"]
+     *     },
+     *     "1": {
+     *       "users": ["XWiki.Admin"]
+     *     }
+     *   ],
+     *   "XWiki.XWikiUsers": 
+     *     "1": {
+     *       "name": ["Spirou"]
+     *     }
+     *   ]
+     * }
+     * 
+     * @param request The input HTTP request that provides the parameters
+     * @return The map containing ordered data (see more details in the method's documentation)
```

Comment:
not sure it's a good idea to refer to the implementation for more details or we could say that everywhere :) I'd rather you remove this text